ptvsd/wrapper.py

```diff
@@ -1864,6 +1866,51 @@ def version_str(v):
         }
         self.send_response(request, **sys_info)
 
+    # VS specific custom message handlers
+    @async_handler
+    def on_setDebuggerProperty(self, request, args):
+        jmc = int(args.get('JustMyCodeStepping', 0))
+        self.debug_options['JUST_MY_CODE'] = jmc > 0
+        self._apply_code_stepping_settings()
+        self.send_response(request)
+
+    def _apply_code_stepping_settings(self):
+        if self.debug_options.get('JUST_MY_CODE', False):
+            project_dirs = []
+            for path in sys.path + [os.getcwd()]:
+                is_stdlib = False
+                norm_path = os.path.normcase(path)
+                for prefix in STDLIB_PATH_PREFIXES:
+                    if norm_path.startswith(prefix):
+                        is_stdlib = True
+                        break
+                if path.endswith('ptvsd') or path.endswith('site-packages'):
+                    is_stdlib = True
+
+                if not is_stdlib and len(path) > 0:
+                    project_dirs.append(path)
+            self.pydevd_notify(pydevd_comm.CMD_SET_PROJECT_ROOTS,
+                               '\t'.join(project_dirs))
+
+    def _is_stdlib(self, filepath):
+        for name in IMPORTLIB_BOOTSTRAP:
+            if filepath.endswith(name):
+                return True
+        for prefix in STDLIB_PATH_PREFIXES:
+            if prefix != '' and filepath.startswith(prefix):
+                return True
+        for name in DONT_TRACE_FILES:
```

Comment:
Why not put `IMPORTLIB_BOOTSTRAP` values into `DONT_TRACE_FILES`?

Reply:
IMPORTLIB_BOOTSTRAP is no longer needed. I will remove that.